pkg/executor/slow_query.go

```diff
@@ -523,32 +522,94 @@ func getLineIndex(offset offset, index int) int {
 	return fileLine
 }
 
-// kvSplitRegex: it was just for split "field: value field: value..."
-var kvSplitRegex = regexp.MustCompile(`\w+: `)
+// findMatchedRightBracket returns the rightBracket index which matchs line[leftBracketIdx]
+// leftBracketIdx should be valid string index for line
+// Returns -1 if invalid inputs are given
+func findMatchedRightBracket(line string, leftBracketIdx int) int {
+	leftBracket := line[leftBracketIdx]
+	rightBracket := byte('}')
+	if leftBracket == '[' {
+		rightBracket = ']'
+	} else if leftBracket != '{' {
+		return -1
+	}
+	current := leftBracketIdx
+	leftBracketCnt := 0
+	for current < len(line) {
+		b := line[current]
+		if b == leftBracket {
+			leftBracketCnt++
+			current++
+		} else if b == rightBracket {
+			leftBracketCnt--
+			if leftBracketCnt > 0 {
+				current++
+			} else if leftBracketCnt == 0 {
+				return current
+			} else {
+				return -1
+			}
+		} else {
+			current++
+		}
+	}
+	return -1
+}
+
+func isLetterOrNumeric(b byte) bool {
+	return ('A' <= b && b <= 'Z') || ('a' <= b && b <= 'z') || ('0' <= b && b <= '9')
+}
 
 // splitByColon split a line like "field: value field: value..."
+// Note:
+// 1. Both field and value string contain only ANSI characters
+// 2. value string may be surrounded by brackets, allowed brackets includes "[]" and "{}",  like {key: value,{key: value}}
+// "[]" can only be nested inside "[]"; "{}" can only be nested inside "{}"
+// 3. value string can't contain ' ' character unless it is inside brackets
```

Comment:
Is origin implementation also contains these restriction?

Reply:
No, while current slow log satisfy these restrictions and it doesn't seem to be changed frequently in future. Previous implementation provides a broader functionality.